packages/muelu/src/Graph/BrickAggregation/MueLu_BrickAggregationFactory_def.hpp

```diff
@@ -400,60 +412,70 @@ namespace MueLu {
   }
 
+ template <class Scalar, class LocalOrdinal, class GlobalOrdinal, class Node>
+  void  BrickAggregationFactory<Scalar, LocalOrdinal, GlobalOrdinal, Node>::getIJK(LocalOrdinal LID, int &i, int &j, int &k) const {
+   i = (*xMap_)[x_[LID]]; 
+   j = (nDim_>1) ? (*yMap_)[y_[LID]] : 0;
+   k =  (nDim_>2) ? (*zMap_)[z_[LID]] : 0;
+  }
+
+
   template <class Scalar, class LocalOrdinal, class GlobalOrdinal, class Node>
-  GlobalOrdinal BrickAggregationFactory<Scalar, LocalOrdinal, GlobalOrdinal, Node>::getAggGID(LocalOrdinal LID) const {
-    // Corrections to ignore Dirichlet BCs
+  void  BrickAggregationFactory<Scalar, LocalOrdinal, GlobalOrdinal, Node>::getAggIJK(LocalOrdinal LID, int &i, int &j, int &k) const {
     int xboost = dirichletX_ ? 1 : 0;
     int yboost = dirichletY_ ? 1 : 0;
     int zboost = dirichletZ_ ? 1 : 0;
-    bool boundary = false;
+    int pointI, pointJ, pointK;
+    getIJK(LID,pointI,pointJ,pointK);
+    i = (pointI-xboost)/bx_;
 
-    int naggx = (nx_-2*xboost)/bx_ + ((nx_-2*xboost) % bx_ ? 1 : 0), naggy = 1;
-    int i = (*xMap_)[x_[LID]], j = nDim_>1 ? (*yMap_)[y_[LID]] : 0, k =  nDim_>2 ? (*zMap_)[z_[LID]] : 0;
-    int ii = (i-xboost)/bx_, jj = 0, kk = 0;
+    if (nDim_ > 1) j = (pointJ-yboost)/by_;
+    else j = 0;
 
-    if( dirichletX_ && (i == 0 || i == nx_ - 1)) boundary = true;
-    if (nDim_ > 1) {
-      naggy = (ny_-2*yboost)/by_ + ( (ny_-2*yboost) % by_ ? 1 : 0);      
-      jj = (j-yboost)/by_;
-      if(dirichletY_ &&  (j == 0 || j == ny_ - 1)) boundary = true;
-    }
+    if (nDim_ > 2) k = (pointK-zboost)/bz_;
+    else k = 0;
+  }
 
-    if (nDim_ > 2) {
-      kk = (k-zboost)/bz_;
-      if( dirichletZ_ && (k == 0 || k == nz_ - 1)) boundary = true;
-    }
+  template <class Scalar, class LocalOrdinal, class GlobalOrdinal, class Node>
+  GlobalOrdinal BrickAggregationFactory<Scalar, LocalOrdinal, GlobalOrdinal, Node>::getAggGID(LocalOrdinal LID) const {
+    bool boundary = false;
+
+    int i, j, k;
+    getIJK(LID,i,j,k);
+    int ii , jj, kk;
+    getAggIJK(LID,ii,jj,kk);
+
+    if(              dirichletX_ && (i == 0 || i == nx_ - 1)) boundary = true;
+    if (nDim_ > 1 && dirichletY_ && (j == 0 || j == ny_ - 1)) boundary = true;
+    if (nDim_ > 2 && dirichletZ_ && (k == 0 || k == nz_ - 1)) boundary = true;
 
     /*
     if(boundary)
-      printf("[%d] coord = (%d,%d,%d) {%d,%d} agg = (%d,%d,%d) {%d,%d} => agg %s\n",LID,i,j,k,nx_,ny_,ii,jj,kk,naggx,naggy,"BOUNDARY");
+      printf("[%d] coord = (%d,%d,%d) {%d,%d,%d} agg = (%d,%d,%d) {%d,%d,%d} => agg %s\n",LID,i,j,k,nx_,ny_,nz_,ii,jj,kk,naggx_,naggy_,naggz_,"BOUNDARY");
     else
-      printf("[%d] coord = (%d,%d,%d) {%d,%d} agg = (%d,%d,%d) {%d,%d} => agg %d\n",LID,i,j,k,nx_,ny_,ii,jj,kk,naggx,naggy,kk*naggy*naggx + jj*naggx + ii);
+      printf("[%d] coord = (%d,%d,%d) {%d,%d,%d} agg = (%d,%d,%d) {%d,%d,%d} => agg %d\n",LID,i,j,k,nx_,ny_,nz_,ii,jj,kk,naggx_,naggy_,naggz_,kk*naggy_*naggx_ + jj*naggx_ + ii);
```

Comment:
missed one?

Reply:
nvm

Reply:
Github has led you astray.  Believe in the heart of the code.